spec/models/product_spec.rb

```diff
@@ -232,7 +209,7 @@ def requires_account?
       end
     end
 
-    context "can_purchase?" do
+    context "can_purchase?", feature_setting: { user_based_price_groups: true } do
```

Comment:
I have a thought on this one since most of the tests in this context still pass even if this is set to `false`.

The `can_purchase?` method doesn't really care about a user's price groups, it just cares about an array of price group ids. What if we ditched the `@user_price_group_ids` variable altogether and use what its value would be if the feature were enabled, `[@price_group.id]`? There are a few places where we create new `PriceGroupMember`s. We don't really need to create those, we could just add its price group to the array.

Thoughts?